### ISSUE TITLE
arrow-c-glib 12.0.1 has "libarrow >=12.0.0,<12.0.1.0a0" dependency

### ISSUE THREAD

**Issue by kou** [conda-forge/arrow-c-glib-feedstock#12]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

arrow-c-glib 12.0.1 has "libarrow >=12.0.0,<12.0.1.0a0" dependency. We can see it by seeing metadata at https://anaconda.org/conda-forge/arrow-c-glib/files .

It means that we can't use arrow-c-glib 12.0.1 with libarrow 12.0.1.

Can we relax the depency to `libarrow >= 12.0.0,<13` or something?

Context: https://github.com/apache/arrow-adbc/issues/859

### Installed packages

```shell
Sorry. I don't have a local environment for this. We have a CI result instead:

https://github.com/apache/arrow-adbc/actions/runs/5395092903/jobs/9818010962#step:6:14

Transaction

  Prefix: /usr/share/miniconda3/envs/test

  Updating specs:

   - cmake
   - compilers
   - gtest[version='>=1.10.0']
   - libpq
   - ninja
   - pkg-config
   - libsqlite
   - pkg-config
   - arrow-c-glib
   - glib
   - gobject-introspection
   - meson
   - ruby
   - arrow-c-glib[version='>=10.0.1']


  Package                           Version  Build               Channel                   Size
─────────────────────────────────────────────────────────────────────────────────────────────────
  Install:
─────────────────────────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex                       0.1  conda_forge         conda-forge/linux-64       3kB
  + _openmp_mutex                       4.5  2_gnu               conda-forge/linux-64      24kB
  + arrow-c-glib                     12.0.1  h977cf35_0          conda-forge/linux-64     782kB
  + aws-c-auth                       0.6.28  hccec9ca_5          conda-forge/linux-64     102kB
  + aws-c-cal                        0.5.27  hf85dbcb_0          conda-forge/linux-64      50kB
  + aws-c-common                     0.8.20  hd590300_0          conda-forge/linux-64     197kB
  + aws-c-compression                0.2.17  h4b87b72_0          conda-forge/linux-64      19kB
  + aws-c-event-stream                0.3.0  hc5de78f_6          conda-forge/linux-64      54kB
  + aws-c-http                        0.7.8  h412fb1b_4          conda-forge/linux-64     193kB
  + aws-c-io                        0.13.26  h0d05201_0          conda-forge/linux-64     154kB
  + pthread-stubs                       0.4  h36c2ea0_1001       conda-forge/linux-64       6kB
  + python                           3.11.4  hab00c5b_0_cpython  conda-forge/linux-64      31MB
  + python_abi                         3.11  3_cp311             conda-forge/linux-64       6kB
  + rdma-core                          28.9  h59595ed_1          conda-forge/linux-64       4MB
  + re2                          2023.03.02  h8c504da_0          conda-forge/linux-64     201kB
  + readline                            8.2  h8228510_1          conda-forge/linux-64     281kB
  + rhash                             1.4.3  h166bdaf_0          conda-forge/linux-64     204kB
  + ruby                              3.2.2  hff50039_0          conda-forge/linux-64       8MB
  + s2n                              1.3.45  h06160fa_0          conda-forge/linux-64     368kB
  + setuptools                       68.0.0  pyhd8ed1ab_0        conda-forge/noarch       464kB
  + snappy                           1.1.10  h9fff704_0          conda-forge/linux-64      39kB
  + sysroot_linux-64                   2.12  he073ed8_15         conda-forge/noarch        33MB
  + tk                               8.6.12  h27826a3_0          conda-forge/linux-64       3MB
  + tzdata                            2023c  h71feb2d_0          conda-forge/noarch       118kB
  + ucx                              1.14.1  hf587318_2          conda-forge/linux-64      17MB
  + wheel                            0.40.0  pyhd8ed1ab_0        conda-forge/noarch        56kB
  + xorg-kbproto                      1.0.7  h7f98852_1002       conda-forge/linux-64      27kB
  + xorg-libice                       1.1.1  hd590300_0          conda-forge/linux-64      58kB
  + xorg-libsm                        1.2.4  h7391055_0          conda-forge/linux-64      27kB
  + xorg-libx11                       1.8.6  h8ee46fc_0          conda-forge/linux-64     829kB
  + xorg-libxau                      1.0.11  hd590300_0          conda-forge/linux-64      14kB
  + xorg-libxdmcp                     1.1.3  h7f98852_0          conda-forge/linux-64      19kB
  + xorg-libxext                      1.3.4  h0b41bf4_2          conda-forge/linux-64      50kB
  + xorg-libxrender                  0.9.10  h7f98852_1003       conda-forge/linux-64      33kB
  + xorg-renderproto                 0.11.1  h7f98852_1002       conda-forge/linux-64      10kB
  + xorg-xextproto                    7.3.0  h0b41bf4_1003       conda-forge/linux-64      30kB
  + xorg-xproto                      7.0.31  h7f98852_1007       conda-forge/linux-64      75kB
  + xz                                5.2.6  h166bdaf_0          conda-forge/linux-64     418kB
  + yaml                              0.2.5  h7f98852_2          conda-forge/linux-64      89kB
  + zlib                             1.2.13  hd590300_5          conda-forge/linux-64      93kB
  + zstd                              1.5.2  h3eb15da_6          conda-forge/linux-64     420kB

  Summary:

  Install: 141 packages

  Total download: 359MB

─────────────────────────────────────────────────────────────────────────────────────────────────



Looking for: ['cmake', 'compilers', "gtest[version='>=1.10.0']", 'libpq', 'ninja', 'pkg-config', 'libsqlite', 'pkg-config', 'arrow-c-glib', 'glib', 'gobject-introspection', 'meson', 'ruby', "arrow-c-glib[version='>=10.0.1']"]


Downloading and Extracting Packages

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
```

### Environment info

```shell
Sorry. I don't have the information.
```

**Comment by kou** [comment]
@h-vetinari may help us.

**Comment by h-vetinari** [comment]
> Can we relax the depency to `libarrow >= 12.0.0,<13` or something?

That already happened with https://github.com/conda-forge/arrow-cpp-feedstock/pull/1087 / https://github.com/conda-forge/arrow-cpp-feedstock/issues/1096 / https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4610.

Now you just need to rebuild arrow-c-glib so it picks up the new metadata (though what you'll get is actually `>=12.0.1,<13.0a0`, now that 12.0.1 is available and will be pulled in during the build).

**Comment by kou** [comment]
Thanks!

**Comment by kou** [comment]
Fixed by #14.
